packages/compass-aggregations/src/modules/auto-preview.ts

```diff
@@ -0,0 +1,41 @@
+import type { AnyAction } from "redux";
+import type { ThunkAction } from "redux-thunk";
+import type { RootState } from ".";
+import { runStage } from "./pipeline";
+
+export enum ActionTypes {
+  AutoPreviewToggled = 'compass-aggregations/autoPreviewToggled',
+}
+
+type AutoPreviewToggledAction = {
+  type: ActionTypes.AutoPreviewToggled;
+};
+
+export const INITIAL_STATE = true;
+
+export default function reducer(state = INITIAL_STATE, action: AnyAction): boolean {
+  if (action.type === ActionTypes.AutoPreviewToggled) {
+    return !state;
+  }
+  return state;
+}
+
+export const toggleAutoPreview = (): ThunkAction<
+  void,
+  RootState,
+  void,
+  AutoPreviewToggledAction
+> => {
+  return (dispatch, getState) => {
+    const {
+      pipeline,
+      autoPreview,
+    } = getState();
+    if (!autoPreview && global?.process?.env?.COMPASS_SHOW_NEW_AGGREGATION_TOOLBAR === 'true') {
+      pipeline.forEach((_stage, index) => dispatch(runStage(index)));
+    }
```

Comment:
`runStage` method will re-run all stages from the index you are passing, not only the stage you are passing, so you don't need to call it for every stage

```suggestion
    if (!autoPreview && global?.process?.env?.COMPASS_SHOW_NEW_AGGREGATION_TOOLBAR === 'true') {
      dispatch(runStage(0))
    }
```